torchquad/integration/newton_cotes.py

```diff
@@ -47,8 +48,16 @@ def calculate_result(self, function_values, dim, n_per_dim, hs):
             backend tensor: Quadrature result
         """
         # Reshape the output to be [N,N,...] points instead of [dim*N] points
-        function_values = function_values.reshape([n_per_dim] * dim)
-
+        self.integrand_shape = function_values.shape[1:]
```

Comment:
There are arguments for the constructor of the class for both integration domain and shape, so we could follow that and add `integrand_shape` as a constructor argument although that seems like it could be inferred as here (I understand the convenience of having it for integration dimension since you can just write the integration domain once and then the number of times/dimensions you use that domain)